interview-questions.md

```diff
@@ -7,52 +7,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What are props in React?
 
-  Your answer:
+  Your answer: Props passes information for parent to child in one direction and cannot be updated.
 
-  Researched answer:
+  Researched answer: In React, Props are arguments that are passed into React components, they are passed to components by way of HTML  attributes. The role of props in  React is similar to JavaScript arguments and attributes in HTML.
 
+https://www.w3schools.com/react/react_props.asp
 
 
-2. What is a DOM event?
+2. What is a DOM event? HTML reacting to HTML events.
 
-  Your answer:
+  Your answer: DOM event are functions executed in order.
 
-  Researched answer:
+  Researched answer: HTML DOM events allow JavaScript to register different event handlers on elements in an HTML document. Events are normally used in combination with functions, and the function will not be executed before the event occurs (such as when a user clicks a button).
 
+https://www.w3schools.com/jsref/dom_obj_event.asp
 
 
 3. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: OOP are objects and the interaction with those objects.
+  
+  OOP stores data in objects, functional programming does not.
 
-  Researched answer:
+  Researched answer: Object-oriented programming (OOP) is programming that is based on the concept of objects. It arranges software design throughout data and objects instead of functions and logic.
 
+  Functional programming and object-oriented programming uses different methods for storing and manipulating the data. In functional programming, data cannot be stored in objects, and it can only be transformed by creating functions. In object-oriented programming, data is stored in objects.
```

Comment:
[Both Functional programming and object-oriented programming uses a different method for storing and manipulating the data. In functional programming, data cannot be stored in objects and it can only be transformed by creating functions. In object-oriented programming, data is stored in objects. The object-oriented programming is widely used by the programmers and successful also.](https://medium.com/@shaistha24/functional-programming-vs-object-oriented-programming-oop-which-is-better-82172e53a526)

The only requirement of the interview section is to write the answers in your own words.